tests/app/main/views/test_suppliers.py

```diff
@@ -1268,53 +1300,130 @@ def test_none_shown_if_user_not_ccs_admin(self, s3, get_signed_url, data_api_cli
         res = self.client.get('/admin/suppliers/1234/agreements/g-cloud-8')
         assert res.status_code == 200
 
-        data = res.get_data(as_text=True)
+        document = html.fromstring(res.get_data(as_text=True))
 
-        assert 'Accept and continue' not in data
-        assert 'Put on hold and continue' not in data
+        assert len(document.xpath("//input[@type='submit'][@value='Accept and continue']")) == 0
```

Comment:
Yes interesting - the main reason I replaced this is because doing a raw substring search is the latter doesn't perform any normalization of the text, and so it's a bad habit to get into.